drivers/i3c/i3c_stm32.c

```diff
@@ -0,0 +1,2447 @@
+/*
+ * Copyright (c) 2024
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/i3c.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <stm32_ll_i3c.h>
+#include <stm32_ll_bus.h>
+#include <stm32_ll_rcc.h>
+#include <stm32h5xx_ll_system.h>
+#include <stm32h5xx_ll_cortex.h>
+#include <zephyr/drivers/pinctrl.h>
+#ifdef CONFIG_I3C_STM32_DMA
+#include <zephyr/drivers/dma/dma_stm32.h>
+#include <zephyr/drivers/dma.h>
+#endif
+#include <zephyr/pm/policy.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(i3c_stm32, CONFIG_I3C_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32_i3c
+
+#define CONFIG_I3C_INIT_PRIORITY 50
```

Comment:
Delete this and make use of already available `CONFIG_I3C_CONTROLLER_INIT_PRIORITY`